src/Uno.PackageDiff/Writers/GitHubWriter.cs

```diff
@@ -0,0 +1,187 @@
+		private readonly HashSet<string> types = new();
+		private readonly string githubPAT;
+		private readonly string sourceRepository;
+		private readonly string githubPRid;
+
+		public GitHubWriter(string githubPAT, string sourceRepository, string githubPRid)
+		{
+			this.githubPAT = githubPAT;
+			this.sourceRepository = sourceRepository;
+			this.githubPRid = githubPRid;
+		}
+
+		public void WriteAssemblyName(string assemblyName) { }
+
+		public void WriteHeader((string Package, NuGetVersion Version) source, (string Package, NuGetVersion Version) target)
+		{
+			this.source = source;
+			this.target = target;
+		}
+
+		public void WriteMissingEvent(EventDefinition eventDef, bool isIgnored)
+		{
+			if(!isIgnored)
+				events.Add(eventDef.ToSignature());
+		}
+		public void WriteMissingEventsEnd() { }
+		public void WriteMissingEventsForTypeEnd() { }
+		public void WriteMissingEventsForTypeStart(string typeName) { }
+		public void WriteMissingEventsStart(int invalidEventsCount) { }
+
+		public void WriteMissingField(FieldDefinition fieldDef, bool isIgnored)
+		{
+			if(!isIgnored)
+				fields.Add(fieldDef.ToSignature());
+		}
+
+		public void WriteMissingFieldForTypeEnd() { }
+		public void WriteMissingFieldForTypeStart(string typeName) { }
+		public void WriteMissingFieldsEnd() { }
+		public void WriteMissingFieldsStart(int invalidFieldsCount) { }
+
+		public void WriteMissingMethod(MethodDefinition methodDef, bool isIgnored)
+		{
+			if(!isIgnored)
+				methods.Add(methodDef.ToSignature());
+		}
+		public void WriteMissingMethodsEnd() { }
+		public void WriteMissingMethodsForTypeEnd() { }
+		public void WriteMissingMethodsForTypeStart(string typeName) { }
+		public void WriteMissingMethodsStart(int invalidMethodsCount) { }
+
+		public void WriteMissingPropertiesEnd() { }
+		public void WriteMissingPropertiesForTypeEnd() { }
+		public void WriteMissingPropertiesForTypeStart(string typeName) { }
+		public void WriteMissingPropertiesStart(int invalidPropertiesCount) { }
+		public void WriteMissingProperty(PropertyDefinition propertyDef, bool isIgnored)
+		{
+			if(!isIgnored)
+				properties.Add(propertyDef.ToSignature());
+		}
+
+		public void WriteMissingType(TypeDefinition typeDef, bool isIgnored)
+		{
+			if(!isIgnored)
+				types.Add(typeDef.ToSignature());
+		}
+		public void WriteMissingTypesEnd() { }
+		public void WriteMissingTypesStart(int invalidTypesCount) { }
+		public void WritePlatform(string platform) { }
+
+		public void Dispose()
+		{
+			using var ms = new System.IO.MemoryStream();
+			using var sw = new System.IO.StreamWriter(ms);
+
+			sw.WriteLine($"# Comparison report for {source.Package} **{source.Version}** with {target.Package} **{target.Version}**");
+			sw.WriteLine();
+
+			if(types.Count + properties.Count + events.Count + methods.Count + fields.Count > 0)
+			{
+				sw.WriteLine("🚨🚨**detected some breaking changes** if changes are wanted, add following line at end of `PackageDiffIgnore.xml`.🚨🚨");
```

Comment:
the sections can be long
you might want to wrap them in \<details> syntax:
<details>
<summary>like so</summary>

~~~xml
<details>
<summary>like so</summary>

```xml
...// note: you need to leave a empty line between <summary> and start of ```
```
</details>
~~~
</details>